recording/lib/storage/s3.ex

```diff
@@ -48,8 +46,23 @@ defmodule Membrane.RTC.Engine.Endpoint.Recording.Storage.S3 do
       |> ExAws.request(aws_config)
 
     case result do
-      {:ok, %{status_code: 200}} -> :ok
-      {:error, response} -> {:error, response}
+      {:ok, %{status_code: 200}} ->
+        :ok
+
+      {:error, response} ->
+        Membrane.Logger.error(
+          "Couldn't save object on S3 bucket, recording id: #{config.recording_id}"
+        )
+
+        {:error, response}
+    end
+  end
+
+  @impl true
+  def on_close(files, recording_id, storage_opts) do
+    case list_objects(recording_id, storage_opts) do
+      {:ok, objects} -> fix_objects(files, recording_id, storage_opts, objects)
+      {:error, :list_objects} -> :error
```

Comment:
suggestion: Maybe we could merge `on_close` with `fix_objects` by using `with`?

Reply:
you cannot use objects in else clause in your proposition

Reply:
You could do something like that:
```elixir
list_objects_result = list_objects(recording_id, storage_opts)
with {:ok, objects} <- list_objects_result
```

Reply:
```elixir
list_objects_result = list_objects(recording_id, storage_opts)
with {:ok, objects} <- list_objects_result,
        :ok <- fix_objects(files, recording_id, storage_opts, objects) do
   :ok
else
   :error -> # can't use object here
```